app.js

```diff
@@ -1,18 +1,52 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const res = require("express/lib/response");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
 
+const SpotifyWebApi = require("spotify-web-api-node");
+
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
 
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
+
+app.get("/artist-search", (req, res, next) => {
+  spotifyApi
+  .searchArtists(req.query.artist)
+  .then((data) => {
+    console.log(data.body.artists);
+    console.log("The received data from the API: ", data.body);
+    res.render("artist-search-results", data)
+  })
```

Comment:
You do not want to send a massive `data `object into view. Instead you should be sending more specific information - what you tried to console.log actually! 
So for example  `res.render("artist-search-results", data.body.artists.items)`